tracer/src/Datadog.Trace/Sampling/TraceSampler.cs

```diff
@@ -55,19 +55,21 @@ public SamplingDecision MakeSamplingDecision(Span span)
         /// <param name="rule">The new rule being registered.</param>
         public void RegisterRule(ISamplingRule rule)
         {
-            for (var i = 0; i < _rules.Count; i++)
+            if (rule is AgentSamplingRule agentSamplingRule)
```

Comment:
AFAICT, we should only try to set `AgentSamplingRule` on a trace sampler _once_, right? Should it be a separate method that handles this explicitly?

Reply:
Done. We now keep only the first rule passed to `RegisterAgentSamplingRule()` and ignore subsequent registrations.